hack/make.sh

```diff
@@ -83,55 +83,50 @@ add_buildtag() {
 	[[ " $DOCKER_BUILDTAGS" == *" $1_"* ]] || DOCKER_BUILDTAGS+=" $1_$2"
 }
 
-if ${PKG_CONFIG} 'libsystemd >= 209' 2> /dev/null; then
-	DOCKER_BUILDTAGS+=" journald"
-elif ${PKG_CONFIG} 'libsystemd-journal' 2> /dev/null; then
-	DOCKER_BUILDTAGS+=" journald journald_compat"
+if [ -z "$CGO_ENABLED" ]; then
+	case "$(go env GOOS)/$(go env GOARCH)" in
+		darwin/* | windows/amd64 | linux/amd64 | linux/arm64 | linux/arm | linux/s390x | linux/ppc64le | linux/riscv*)
+			export CGO_ENABLED=1
+			;;
+		*)
+			export CGO_ENABLED=0
+			;;
+	esac
 fi
 
-# test whether "libdevmapper.h" is new enough to support deferred remove
-# functionality. We favour libdm_dlsym_deferred_remove over
-# libdm_no_deferred_remove in dynamic cases because the binary could be shipped
-# with a newer libdevmapper than the one it was built with.
-if
-	command -v gcc &> /dev/null \
-		&& ! (echo -e '#include <libdevmapper.h>\nint main() { dm_task_deferred_remove(NULL); }' | gcc -xc - -o /dev/null $(pkg-config --libs devmapper) &> /dev/null) \
-		;
-then
-	add_buildtag libdm dlsym_deferred_remove
+if [ "$CGO_ENABLED" = "1" ] && [ "$DOCKER_LINKMODE" = "static" ] && [ "$(go env GOOS)" = "linux" ]; then
+	DOCKER_LDFLAGS+=" -extldflags -static"
 fi
 
-# Use these flags when compiling the tests and final binary
-
-IAMSTATIC='true'
-if [ -z "$DOCKER_DEBUG" ]; then
-	LDFLAGS='-w'
+if [ "$CGO_ENABLED" = "1" ] && [ "$DOCKER_LINKMODE" = "static" ]; then
+	DOCKER_BUILDTAGS+=" netgo osusergo static_build"
 fi
 
-LDFLAGS_STATIC=''
-EXTLDFLAGS_STATIC='-static'
-# ORIG_BUILDFLAGS is necessary for the cross target which cannot always build
-# with options like -race.
-ORIG_BUILDFLAGS=(-tags "netgo osusergo static_build $DOCKER_BUILDTAGS" -installsuffix netgo)
-# see https://github.com/golang/go/issues/9369#issuecomment-69864440 for why -installsuffix is necessary here
-
-BUILDFLAGS=(${BUILDFLAGS} "${ORIG_BUILDFLAGS[@]}")
-
-LDFLAGS_STATIC_DOCKER="
-	$LDFLAGS_STATIC
-	-extldflags \"$EXTLDFLAGS_STATIC\"
-"
+if [ "$CGO_ENABLED" = "1" ] && [ "$(go env GOOS)" != "windows" ] && [ "$DOCKER_LINKMODE" != "static" ]; then
+	# pkcs11 cannot be compiled statically if CGO is enabled (and glibc is used)
+	DOCKER_BUILDTAGS+=" pkcs11"
+fi
 
-if [ "$(uname -s)" = 'FreeBSD' ]; then
-	# Tell cgo the compiler is Clang, not GCC
-	# https://code.google.com/p/go/source/browse/src/cmd/cgo/gcc.go?spec=svne77e74371f2340ee08622ce602e9f7b15f29d8d3&r=e6794866ebeba2bf8818b9261b54e2eef1c9e588#752
-	export CC=clang
+if ${PKG_CONFIG} 'libsystemd >= 209' 2> /dev/null; then
+	DOCKER_BUILDTAGS+=" journald"
+elif ${PKG_CONFIG} 'libsystemd-journal' 2> /dev/null; then
+	DOCKER_BUILDTAGS+=" journald journald_compat"
+fi
```

Comment:
I'm actually wondering if we still need this at all. Looks like this was added in https://github.com/moby/moby/commit/6cdc4ba6cd5178037466c50ebe03a7eb111c43b1 (https://github.com/moby/moby/pull/20617) in 2016, and likely because at the time we were still building for CentOS 6.

doing some quick checks;

```bash
docker run -it --rm centos:7

yum install -y systemd-devel

pkg-config 'libsystemd >= 209' && echo "OK" || echo "KO"
OK

pkg-config --print-provides 'libsystemd'
libsystemd = 219

pkg-config --print-provides 'libsystemd-journal'
libsystemd-journal = 219
```

And on a `debian:buster` (oldstable)

```bash
docker run -it --rm debian:buster

apt-get update && apt-get install -y libsystemd-dev pkg-config

pkg-config 'libsystemd >= 209' && echo "OK" || echo "KO"
OK

pkg-config --print-provides 'libsystemd'
libsystemd = 241

# only for old distros (now part of libsystemd-dev); https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=795147
pkg-config --print-provides 'libsystemd-journal'
Package libsystemd-journal was not found in the pkg-config search path.
Perhaps you should add the directory containing `libsystemd-journal.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libsystemd-journal' found
```

OpenSUSE leap (I think that's built for s390x)

```bash
docker run -it --rm docker.io/opensuse/leap:15

zypper install -y systemd-devel

pkg-config 'libsystemd >= 209' && echo "OK" || echo "KO"
OK

pkg-config --print-provides 'libsystemd'
libsystemd = 246

pkg-config --print-provides 'libsystemd-journal'
Package libsystemd-journal was not found in the pkg-config search path.
Perhaps you should add the directory containing `libsystemd-journal.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libsystemd-journal' found
```